orttraining/orttraining/python/training/ortmodule/configuration/_configuration_option.py

```diff
@@ -0,0 +1,16 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# _configuration_option.py
+
+from abc import ABC, abstractmethod
+
+class ConfigurationOption(ABC):
+    """All configurable options should inherit from this base class."""
+
+    @abstractmethod
+    def configure(self):
+        pass
+
+    @abstractmethod
+    def reset(self):
+        pass
```

Comment:
Could you add this to the design document for discussion?

Just by looking the UT I felt this API is very verbose for the final user. It was required three long lines to pass little information

It doesn't feel a PyTorch API but more like a low level C API :)

```
        debug_options= DebugOptions()
        debug_options.save_intermediate_onnx_models.configure(save=False, prefix='MNIST')
        debug_options.logging.configure(loglevel=LogLevel.VERBOSE)

        model = ORTModule(model, debug_options)
```

A simpler alternative could be 
```
model = ORTModule(model, DebugOptions(log_level=LogLevel.VERBOSE, save_onnx=True)
```
which assumes the ONNX model will be saved in the current working directory. If the user is not happy about the default path, it change it by

```
os.environ["ORTMODULE_SAVE_ONNX_PATH"] = "/opt/save_here/"

```

Another alternative, which is semantically poorer is
```
model = ORTModule(model, DebugOptions(log_level=LogLevel.VERBOSE, save_onnx_path='/opt/')
```

that assumes '' or None disables model saving

Reply:
The problem with this approach is that configuration flags could have multiple parameters (such as ```SaveOnnx```).

We probably want grouping. For example, this code snippet:
```py
model = ORTModule(model, DebugOptions(log_level=LogLevel.VERBOSE, save_onnx=True, save_onnx_prefix='my_model', dst_folder='my/path')
```
seems not modular or easy to decipher.  It is possible that in the future we have even more options in this ```DebugOptions``` class that need grouping.

The idea of having only the environment variable control the location of the onnx models does not seem very intuitive either. If it were offered in addition to a "Pythonic" API way of doing this, it would make sense. 

I'll add this to the design doc for discussion.

Reply:
Going with the approach:
```py
model = ORTModule(model, DebugOptions(log_level=LogLevel.VERBOSE, save_onnx=True, save_onnx_prefix='my_model')
```

User can override the log level by setting the environment variable: ```ORTModuleLogLevel```. Defaults to ```LogLevel.WARNING```
User can override the save onnx destination directory by setting the variable: ```ORTModuleSaveONNXDirectory```. Defaults to current working directory.